### PR TITLE
Improve code coverage for encode

### PR DESCRIPTION
**Pull request by Ja-Gk-00**
Added 3 unit tests that cover some edge cases for encoder to increase test code coverage.

### CODE REVIEW

File: tests/unit_tests/test_serialize/test_lookups/test_term_encoder.py [hunk 1]
```diff
@@ -54,3 +73,13 @@ def test_encode_literal_ok_with_string_and_langtag(subtests: SubTests) -> None:
         assert literal.lex == snapshot("baz")
         assert literal.langtag == snapshot("en")
         assert literal.datatype == snapshot(0)
+
+
+def test_encode_namespace_declaration() -> None:
```
Comment (adanilenka) [inline tests/unit_tests/test_serialize/test_lookups/test_term_encoder.py:78]: this is ok, can we add more tests for namespace declaration in general?

File: tests/unit_tests/test_serialize/test_lookups/test_term_encoder.py [hunk 2]
```diff
@@ -2,9 +2,21 @@
...
-from pyjelly.serialize.encode import TermEncoder
+from pyjelly.serialize.encode import (
+    Slot,
+    TermEncoder,
+    encode_namespace_declaration,
+    split_iri,
+)
+
+
+def test_split_iri_no_sep() -> None:
```
[the remark this answers is not in the document]
Comment (Ja-Gk-00) [reply tests/unit_tests/test_serialize/test_lookups/test_term_encoder.py]: Ok, changed the split_IRI so that it raises for invalid data; however, I still need to return tuple[str, str] so there is an unreachable one-line of code so that mypy passes :<.
But in general, it's fine, we don't have to go for total 100% cover, because turning off mypy for one function is excessive.

**Review by adanilenka — APPROVED** [review]
